lib/sarif/sarif_report.rb

```diff
@@ -0,0 +1,56 @@
+require 'json'
+require 'json-schema'
+require_relative './base_sarif'
+require_relative './gosec_sarif'
+
+module Sarif
+  # Class for generating sarif reports
+  class SarifReport
+    class SarifInvalidFormatError < StandardError; end
+
+    SARIF_VERSION = "2.1.0".freeze
+
+    SARIF_SCHEMA = "https://schemastore.azurewebsites.net/schemas"\
+      "/json/sarif-2.1.0-rtm.5.json".freeze
+
+    def initialize(scan_reports)
+      @scan_reports = scan_reports
+    end
+
+    # Builds Sarif Report
+    #
+    # @return [JSON]
+    def to_sarif
+      sarif_report = {
+        "version" => SARIF_VERSION,
+        "$schema" => SARIF_SCHEMA,
+        "runs" => []
+      }
+      # for each scanner report, run the appropriate converter
+      @scan_reports.each { |scan_report| sarif_report["runs"] << converter(scan_report[0]) }
+      report = JSON.pretty_generate(sarif_report)
+      path = File.expand_path('schema/sarif-schema.json', __dir__)
+      schema = JSON.parse(File.read(path))
+
+      if JSON::Validator.validate(schema, report)
```

Comment:
Is it worth indicating how validation failed? E.g.

`#`
`# return an array of error messages when validation fails`
`#`
`JSON::Validator.fully_validate(schema, { "a" => "taco" })`

Reply:
Yes, information about the error will be helpful. I've added some documentation for this